ludwig/datasets/loaders/av_mnist.py

```diff
@@ -0,0 +1,214 @@
+                    self.mnist_data = np.load(os.path.join(self.raw_dataset_dir, "image", "train_data.npy"))
+                    self.labels = np.load(os.path.join(self.raw_dataset_dir, "train_labels.npy"))
+                else:
+                    self.audio_data = np.load(os.path.join(self.raw_dataset_dir, "audio", "test_data.npy"))
+                    self.mnist_data = np.load(os.path.join(self.raw_dataset_dir, "image", "test_data.npy"))
+                    self.labels = np.load(os.path.join(self.raw_dataset_dir, "test_labels.npy"))
+
+                self.audio_data = self.audio_data[:, np.newaxis, :, :]
+                self.mnist_data = self.mnist_data.reshape(self.mnist_data.shape[0], 1, 28, 28)
+                self.write_output_dataset(self.labels, self.audio_data, os.path.join(self.raw_dataset_dir, dataset))
+                self.write_output_dataset(self.labels, self.mnist_data, os.path.join(self.raw_dataset_dir, dataset))
+            else:
+                if self.modal:
+                    if self.modal not in ["audio", "image"]:
+                        raise ValueError("the value of modal is allowed")
+                    if dataset == "train":
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "train_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "train_labels.npy"))
+                    else:
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "test_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "test_labels.npy"))
+
+                    if self.modal == "audio":
+                        self.data = self.data[:, np.newaxis, :, :]
+                    elif self.modal == "image":
+                        self.data = self.data.reshape(self.data.shape[0], 1, 28, 28)
+                    self.write_output_dataset(self.labels, self.data, os.path.join(self.raw_dataset_dir, dataset))
+                else:
+                    raise ValueError("the value of modal should be given")
+
+        return super().transform_files(file_paths)
+
+    def load_unprocessed_dataframe(self, file_paths: List[str]) -> pd.DataFrame:
+        """Load dataset files into a dataframe."""
+        return self.output_training_and_test_data()
+
+    def process_source_dataset(self, raw_dataset_dir="."):
+        """An intermediate method to write .npy files for the training and the test image and label data to the raw dataset directory
+        :args:
+            raw_dataset_dir (str) : the directory where all the raw data lives
+        :returns:
+            None"""
+        file_names = {
+            "train_data": "train-images-idx3-ubyte.gz",
+            "train_labels": "train-labels-idx1-ubyte.gz",
+            "test_data": "t10k-images-idx3-ubyte.gz",
+            "test_labels": "t10k-labels-idx1-ubyte.gz",
+        }
```

Comment:
nit: use `self.file_names`

Reply:
Actually, you can remove this completely since you're already using `self.file_names.items()` in like 130

Reply:
removed , we're using self.config object to retrieve this dictionary from the yaml file